build_tooling/transform_asv_results.py

```diff
@@ -0,0 +1,160 @@
+"""
+Copyright 2024 Man Group Operations Limited
+
+Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+
+As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+"""
+import pandas as pd
+from arcticdb.storage_fixtures.s3 import real_s3_from_environment_variables
+import json
+from pathlib import Path
+from arcticdb import Arctic
+from argparse import ArgumentParser
+
+
+def setup_machine_folder(json_data, machine_path):
+    machine_path.mkdir()
+    with open(machine_path / "machine.json", "w") as out:
+        machine_data = json_data["params"]
+        machine_data["version"] = 1
+        json.dump(machine_data, out, indent=4, default=str)
+
+
+def df_to_asv_json(results_df: pd.DataFrame):
+    new_df = results_df.copy()
+    new_df["date"] = (pd.to_datetime(results_df["date"]).astype(int) // 1000000).astype(
+        object
+    )
+    new_df["version"] = results_df["version"].astype(object)
+
+    metadata = {
+        "commit_hash": new_df["commit_hash"].iloc[0],
+        "env_name": new_df["env_name"].iloc[0],
+        "date": new_df["date"].iloc[0],
+        "params": eval(new_df["params"].iloc[0]),
```

Comment:
Added example.

`json.loads` doesn't work because they are not valid jsons list/dicts because they contain mixed types, e.g. lists with ints and float.
That is why we need to use eval to evaluate them to Python lists/dicts